components/dashboard-modules/EnergyReccomendation.tsx

```diff
@@ -0,0 +1,71 @@
+import Image from 'next/image';
+import { FC } from 'react';
+import { useSiteData } from '~/lib/hooks';
+import { getCurrentTimeForecast } from '~/lib/utils';
+import content from '../../content/power-card-content.json';
+import NumberDisplay from './NumberDisplay';
+
+/**
+ * Determines the appliance with the greatest energy required that is less than or equal to the current output
+ * @param currentOutput the current output in kW based on the getCurrentTimeForecast
+ * @returns the index of the appliance with the greatest enrgy requiement less than or equal to the current ouput
+ * or -1 if an index could not be found
+ */
+
+const getBestRecomendationIndex = (currentOutput: number | undefined) => {
+  if (currentOutput != undefined) {
+    let maxIndex = -1;
+    let maxKW = 0;
+    for (let i = 0; i < content.appliances.length; i++) {
+      const currAppliance = content.appliances[i];
+      if (
+        Number(currAppliance.kW) > maxKW &&
+        Number(currAppliance.kW) <= currentOutput
+      ) {
+        maxIndex = i;
+        maxKW = Number(currAppliance.kW);
+      }
+    }
+    return maxIndex;
+  }
+  return -1;
+};
+
+const EnergyRecommendation: FC<{ siteUUID: string }> = ({ siteUUID }) => {
+  const { forecastData, isLoading } = useSiteData(siteUUID);
+  const currentOutput = forecastData
+    ? getCurrentTimeForecast(forecastData.forecast_values)
+    : undefined;
+  const reccomendationIdx = getBestRecomendationIndex(currentOutput);
```

Comment:
a few typos here
```suggestion
  const recommendationIdx = getBestRecommendationIndex(currentOutput);
```
also, would probably be good to just have `recommendationIndex` defined as a ternary 
```suggestion
  const recommendationIdx = currentOutput ? getBestRecommendationIndex(currentOutput) : undefined
```
this will allow us to avoid having to check for undefinedness in getBestRecommendationIndex, because I feel like we shouldn't have to worry about checking for undefinedness in that function.